src/draw/shapes.typ

```diff
@@ -555,10 +561,17 @@
       close: close
     )
 
+    // Find center for simple polygons, might return none
+    let centroid = if close {
+      polygon.simple-centroid(pts)
+    }
```

Comment:
could this be done within the named anchor function? it would then be cached by typst when its needed instead of always